# Proyecto de Programación de Algoritmos

Nombre: José Pullaguari

In [1]:
#Importamos para trabajar con la libreria pandas
import pandas as pd

#### Presentación de todos los datos

In [2]:
#Usemos la funcion de pandas read_csv para cargar los datos. Esto crea un DataFrame, una unidad de datos en Pandas.
datos = pd.read_csv("data/nacimientos2014.csv", sep=",")
#Presentamos los datos consumidos 
datos 

Unnamed: 0            tip_insc  ofi_insc  \
0              1  Inscripcion Normal         1   
1              2  Inscripci�n Normal         1   
2              3  Inscripci�n Normal         1   
3              4  Inscripci�n Normal         1   
4              5  Inscripci�n Normal         1   
5              6  Inscripci�n Normal         1   
6              7  Inscripci�n Tard�a         1   
7              8  Inscripci�n Normal         1   
8              9  Inscripci�n Tard�a         1   
9             10  Inscripci�n Normal         1   
10            11  Inscripci�n Tard�a         1   
11            12  Inscripci�n Normal         1   
12            13  Inscripci�n Normal         1   
13            14  Inscripci�n Normal         1   
14            15  Inscripci�n Normal         1   
15            16  Inscripci�n Normal         1   
16            17  Inscripci�n Normal         1   
17            18  Inscripci�n Normal         1   
18            19  Inscripci�n Tard�a         1   
19            20  Inscripci�n Normal         1   
20            21  Inscripci�n Normal         1   
21            22  Inscripci�n Normal         1   
22            23  Inscripci�n Normal         1   
23            24  Inscripci�n Normal         1   
24            25  Inscripci�n Normal         1   
25            26  Inscripci�n Normal         1   
26            27  Inscripci�n Tard�a         1   
27            28  Inscripci�n Normal         1   
28            29  Inscripci�n Normal         1   
29            30  Inscripci�n Normal         1   
...          ...                 ...       ...   
2455        2456  Inscripci�n Normal         1   
2456        2457  Inscripci�n Tard�a         1   
2457        2458  Inscripci�n Normal         1   
2458        2459  Inscripci�n Normal         1   
2459        2460  Inscripci�n Normal         1   
2460        2461  Inscripci�n Normal         1   
2461        2462  Inscripci�n Normal         1   
2462        2463  Inscripci�n Normal         1   
2463        2464  Inscripci�n Normal         1   
2464        2465  Inscripci�n Normal         1   
2465        2466  Inscripci�n Tard�a         1   
2466        2467  Inscripci�n Normal         1   
2467        2468  Inscripci�n Normal         1   
2468        2469  Inscripci�n Tard�a         1   
2469        2470  Inscripci�n Normal         1   
2470        2471  Inscripci�n Normal         1   
2471        2472  Inscripci�n Tard�a         1   
2472        2473  Inscripci�n Tard�a         1   
2473        2474  Inscripci�n Normal         1   
2474        2475  Inscripci�n Tard�a         2   
2475        2476  Inscripci�n Normal         1   
2476        2477  Inscripci�n Normal         1   
2477        2478  Inscripci�n Normal         1   
2478        2479  Inscripci�n Normal         1   
2479        2480  Inscripci�n Normal         1   
2480        2481  Inscripci�n Normal         1   
2481        2482  Inscripci�n Normal         1   
2482        2483  Inscripci�n Normal         1   
2483        2484  Inscripci�n Normal         1   
2484        2485  Inscripci�n Normal         1   

                           prov_insc                cant_insc  \
0                             Guayas                  Milagro   
1                             Guayas                  Milagro   
2                             Guayas                  Milagro   
3                             Guayas  San Jacinto de Yaguachi   
4                             Guayas                  Milagro   
5                             Guayas                  Milagro   
6                             Guayas                    Daule   
7                             Guayas                    Daule   
8                             Guayas                    Daule   
9                             Guayas                  Milagro   
10                         Pichincha                  Cayambe   
11                            Guayas                    Daule   
12                            Guayas                    Daul

### Presentamos la cantidad de nacidos por Sexos

In [3]:
#Importamos show para presentar las graficas y el output_notebook para visualizarlas las celdas del jupyter
from bokeh.io import show, output_notebook
#El ColumnDataSource para compartir datos entre múltiples gráficos y widgets 
from bokeh.models import ColumnDataSource, HoverTool
#Se encargan de armar gráficos con ejes, cuadrículas y herramientas predeterminadas.
from bokeh.plotting import figure
#"Spectral6" proporciona una lista de seis elementos de cadenas hexagonales RGB (A) del mapa de colores "Spectral" de Brewer.
from bokeh.palettes import Spectral6
#Asigna colores a elementos categóricos específicos
from bokeh.transform import factor_cmap


#Presentamos la cantidad de mujeres y hombres nacidos en el 2014
sexo = list(datos["sexo"])#Creamos una lista con que contiene todos los sexos
sexIndex = list(set(sexo))#Asigamos la lista a otra variable sexIndex

#Contamos el numero de cada sexo por categorias con la funcion .count() y almacenamos en variable d1 y d2
d1 = sexo.count(sexIndex[0])
d2 = sexo.count(sexIndex[1])

pos_x = sexIndex#Datos que van en el eje x que son los dos tipos de sexo
pos_y = [d1,d2]#Datos que van en el eje y que son las cantidades por sexo contadas

#ColumnDataSource es el objeto donde se almacenan los datos de un gráfico Bokeh. 

#Colocamos todos los datos, en un ColumnDataSource. 
source = ColumnDataSource(data = dict(pos_x=pos_x, pos_y=pos_y))


#llamamos al metodo para presentar en celdas de jupyter solo una vez
output_notebook()

#Figure principal clase de plotting, esto crea un Figure modelo, que incluye métodos para agregar diferentes tipos de glifos a un gráfico.
graf_Sexos = figure(x_range=pos_x, plot_height=350, toolbar_location=None, title="Cantidades por Sexos")


#Utilizamo el vbar y establecemos la coordenada X y el top que es el alto de las barras
#el nombre de la columna Source se pasa a figure
#Usamos los métodos de glifos .vbar para agregar renderizadores para nuestros datos 
graf_Sexos.vbar(x='pos_x',  top='pos_y', width=0.4, source=source, legend="pos_y",
       line_color='black', fill_color = factor_cmap('pos_x', palette = Spectral6, factors=pos_x))


graf_Sexos.xgrid.grid_line_color = None     #Quitamos la linea vertical
graf_Sexos.y_range.start = 0                #Comenzamos el eje y en 0
graf_Sexos.y_range.end = 1500               #Terminamos el eje y en 1500
graf_Sexos.legend.orientation = "horizontal"#Indicadores con cantidades
graf_Sexos.legend.location = "top_right"     #Localizacion del indicador

#Funcion para presentar
show(graf_Sexos)

Loading BokehJS ...

En esta gráfica se presenta la contabilización de nacimientos por sexo donde las dos cantidades dan 2485 que es el número de registros de nuestro csv, por lo cual esta correcta esa cantidad, además se puede observar que en el 2014 el sexo con más nacimientos son los hombres

### Presentamos la cantidad de nacidos por cada mes

#### En esta gráfica se muestra el total de nacimientos por cada mes durante el año 2014, se puede obsevar que el mes de julio es el que contiene mayor nacimientos

In [4]:
#Presentamos la cantidad de nacidos por cada mes

meses = list(datos["mes_nac"])#Creamos una lista con que contiene todos los meses
posMes = list(set(meses))#Asigamos la lista a otra variable posMes

#Contamos con la funcion .count() numero de nacimientos por meses y se asigna a cada variable una posición
mes1 = meses.count(posMes[0])
mes2 = meses.count(posMes[1])
mes3 = meses.count(posMes[2])
mes4 = meses.count(posMes[3])
mes5 = meses.count(posMes[4])
mes6 = meses.count(posMes[5])
mes7 = meses.count(posMes[6])
mes8 = meses.count(posMes[7])
mes9 = meses.count(posMes[8])
mes10 = meses.count(posMes[9])
mes11 = meses.count(posMes[10])
mes12 = meses.count(posMes[11])


#Colores para las barras de cada mes
colores=["#C1C83C","#97AF71","#F61F46","#4A56F7","#D66F3C","#9890A9","#F1B3C2","#9FEE71","#64E7E7","#71AF7E","#A0A4A9","#51EAA7"]
#Almacenamos en una lista todos variables que tienen el numero de nacimientos de cada mes
lista_meses=[mes1,mes2,mes3,mes4,mes5,mes6,mes7,mes8,mes9,mes10,mes11,mes12]

#Creamos objeto graf_meses -  establecemos coordenadas del eje X y Y
graf_meses = figure(x_range= posMes, plot_width=1000,plot_height=300, title="Cantidad de nacimientos por mes", toolbar_location="below", tools="")
graf_meses.vbar(x=posMes, top = lista_meses , width=0.6, color=colores)

graf_meses.xgrid.grid_line_color= None#Quitamos la linea vertical
graf_meses.y_range.start=0            #Comenzamos el eje y en 0

show(graf_meses )#Presentamos la grafica

In [5]:
#Cantidades obtenidas
mes1,mes2,mes3,mes4,mes5,mes6,mes7,mes8,mes9,mes10,mes11,mes12

(147, 89, 224, 218, 206, 221, 302, 163, 307, 90, 379, 139)

Para visualizar de mejor manera todos los atributos del csv consumido ingresamos "datos.columns" 

In [6]:
datos.columns

Index(['Unnamed: 0', 'tip_insc', 'ofi_insc', 'prov_insc', 'cant_insc',
       'parr_insc', 'anio_insc', 'mes_insc', 'dia_insc', 'fecha_insc', 'sexo',
       'talla', 'peso', 'sem_gest', 'tipo_part', 'anio_nac', 'mes_nac',
       'dia_nac', 'fecha_nac', 'p_emb', 'lugar_ocur', 'prov_nac', 'cant_nac',
       'parr_nac', 'area_nac', 'asis_por', 'nac_mad', 'cod_pais', 'anio_mad',
       'mes_mad', 'dia_mad', 'fecha_mad', 'edad_mad', 'con_pren', 'num_par',
       'hij_viv', 'hij_vivm', 'hij_nacm', 'etnia', 'est_civil', 'sabe_leer',
       'niv_inst', 'prov_res', 'cant_res', 'parr_res', 'area_res'],
      dtype='object')

Ahora colocamos como indice al atributo "tipo_part", de esta manera identificamos los tipos de parto que tiene el csv para poder graficar

In [7]:
#Se establece tipos de parto como index
datos.set_index('tipo_part')

Unnamed: 0            tip_insc  ofi_insc  \
tipo_part                                             
Ces�rea             1  Inscripcion Normal         1   
Ces�rea             2  Inscripci�n Normal         1   
Ces�rea             3  Inscripci�n Normal         1   
Ces�rea             4  Inscripci�n Normal         1   
Normal              5  Inscripci�n Normal         1   
Ces�rea             6  Inscripci�n Normal         1   
Normal              7  Inscripci�n Tard�a         1   
Ces�rea             8  Inscripci�n Normal         1   
NaN                 9  Inscripci�n Tard�a         1   
Normal             10  Inscripci�n Normal         1   
NaN                11  Inscripci�n Tard�a         1   
Ces�rea            12  Inscripci�n Normal         1   
Normal             13  Inscripci�n Normal         1   
Ces�rea            14  Inscripci�n Normal         1   
Ces�rea            15  Inscripci�n Normal         1   
Ces�rea            16  Inscripci�n Normal         1   
Ces�rea            17  Inscripci�n Normal         1   
Normal             18  Inscripci�n Normal         1   
NaN                19  Inscripci�n Tard�a         1   
Normal             20  Inscripci�n Normal         1   
Ces�rea            21  Inscripci�n Normal         1   
Normal             22  Inscripci�n Normal         1   
Normal             23  Inscripci�n Normal         1   
Ces�rea            24  Inscripci�n Normal         1   
Ces�rea            25  Inscripci�n Normal         1   
Normal             26  Inscripci�n Normal         1   
NaN                27  Inscripci�n Tard�a         1   
Normal             28  Inscripci�n Normal         1   
Ces�rea            29  Inscripci�n Normal         1   
Ces�rea            30  Inscripci�n Normal         1   
...               ...                 ...       ...   
Ces�rea          2456  Inscripci�n Normal         1   
Ces�rea          2457  Inscripci�n Tard�a         1   
Ces�rea          2458  Inscripci�n Normal         1   
Ces�rea          2459  Inscripci�n Normal         1   
Ces�rea          2460  Inscripci�n Normal         1   
Ces�rea          2461  Inscripci�n Normal         1   
Normal           2462  Inscripci�n Normal         1   
Ces�rea          2463  Inscripci�n Normal         1   
Ces�rea          2464  Inscripci�n Normal         1   
Normal           2465  Inscripci�n Normal         1   
Normal           2466  Inscripci�n Tard�a         1   
Normal           2467  Inscripci�n Normal         1   
Ces�rea          2468  Inscripci�n Normal         1   
Ces�rea          2469  Inscripci�n Tard�a         1   
Ces�rea          2470  Inscripci�n Normal         1   
Ces�rea          2471  Inscripci�n Normal         1   
Ces�rea          2472  Inscripci�n Tard�a         1   
Ces�rea          2473  Inscripci�n Tard�a         1   
Normal           2474  Inscripci�n Normal         1   
Normal           2475  Inscripci�n Tard�a         2   
Ces�rea          2476  Inscripci�n Normal         1   
Normal           2477  Inscripci�n Normal         1   
Ces�rea          2478  Inscripci�n Normal         1   
Normal           2479  Inscripci�n Normal         1   
Ces�rea          2480  Inscripci�n Normal         1   
Ces�rea          2481  Inscripci�n Normal         1   
Ces�rea          2482  Inscripci�n Normal         1   
Normal           2483  Inscripci�n Normal         1   
Normal           2484  Inscripci�n Normal         1   
Ces�rea          2485  Inscripci�n Normal         1   

                                prov_insc                cant_insc  \
tipo_part                                                            
Ces�rea                            Guayas                  Milagro   
Ces�rea                            Guayas                  Milagro   
Ces�rea                            Guayas                  Milagro   
Ces�rea                            Guayas  San Jacinto de Yaguachi   
Normal                             Guayas                  Milagro   
Ces�rea                            Guayas              

Se identifico 3 tipos de Parto:
        - Normal
        - Cesárea
        - Ignorado

In [8]:
#Filtramos los tipos de parto Normal y almacenamos en la variable normal
normal = datos[datos['tipo_part']=='Normal']
#Contamos el numero de normal y presentamos un encabezado para ver su numero
normal.count().head(1)

Unnamed: 0    813
dtype: int64

In [9]:
#Filtramos los tipos de parto Cesarea y almacenamos en cesarea
cesarea = datos[datos['tipo_part']=='Ces�rea']
#Contamos el numero de cesarea y presentamos un encabezado para ver su numero
cesarea.count().head(1)

Unnamed: 0    1044
dtype: int64

In [10]:
#Filtramos los tipos de parto Ignorado y almacenamos en ignora
ignora = datos[datos['tipo_part']=='Ignorado']
#Contamos el numero de ignora y presentamos un encabezado para ver su numero
ignora.count().head(1)

Unnamed: 0    44
dtype: int64

### Presentamos datos por cada tipo de parto establecido

In [11]:
#Libreria a utilizar para el tipo de grafia elegido
from bokeh.palettes import OrRd3 #Serie de 3  colores para las barras

#Establecemos la etiqueta a graficar
tipos = ['Tipo:  NORMAL,  CESÁREA,  IGNORADO']
#Categorias de tipo de parto
categorias = ["Normal", "Cesárea","Ignorado"]

#Colocamos cada valor obtenido anteriormente con cada categoria
dataT = {'tipos' : tipos,
           'Normal'   : [813],
           'Cesárea'   : [1044],
           'Ignorado':[44]}

#Creamos objeto graf_TiposP -  establecemos coordenadas del eje X y Y , rango de y solo un tipo, rango x.
graf_TiposP = figure(y_range = tipos, plot_height = 250, x_range = (0, 2000), title = "Número Tipos de Partos en el 2014",
           toolbar_location=None)

#Tipo de glifo hbar_stack, establecemos sus caracteristicas de la barra
graf_TiposP.hbar_stack(categorias, y = 'tipos', height = 0.9, color = OrRd3, source = ColumnDataSource(dataT),
             legend = ["%s" % x for x in categorias])

graf_TiposP.y_range.range_padding = 0.1          #relleno de la forma
graf_TiposP.ygrid.grid_line_color = None         #quitamos la linea horizontal
graf_TiposP.legend.location = "top_left"         #colocamos los indicadores en la izquierda superior
graf_TiposP.axis.minor_tick_line_color = "black" #color de lineas de intervalo
graf_TiposP.outline_line_color =  None           #linea que rodea toda la grafica

show(graf_TiposP)

#### En esta gráfica se puede visualizar que hay más nacimientos que nacieron por Cesáreas durante el año 2014

## Provincias de la Sierra
Se ha escogido las provincias en la región Sierra con mayor cantidad de nacimientos inscritos 

In [12]:
#Se escoge una provincia 1
prov1=datos[datos['prov_insc']=='Pichincha']
#Presentamos el numero 
prov1.count().head()

Unnamed: 0    347
tip_insc      347
ofi_insc      347
prov_insc     347
cant_insc     347
dtype: int64

In [13]:
#Se escoge una provincia 2
prov2=datos[datos['prov_insc']=='Azuay']
prov2.count().head()

Unnamed: 0    90
tip_insc      90
ofi_insc      90
prov_insc     90
cant_insc     90
dtype: int64

In [14]:
#Se escoge una provincia 3
prov3=datos[datos['prov_insc']=='Imbabura']
prov3.count().head()

Unnamed: 0    86
tip_insc      86
ofi_insc      86
prov_insc     86
cant_insc     86
dtype: int64

## Gráfica con datos de la provincia de la Sierra con más nacimientos inscritos, detallando por asistentes del parto

Presentamos los datos de la provincia Pichincha que es la que tiene más nacimientos inscritos con el numero de asistentes que intervinieron en los partos con categorías de:

            - Médico/a
            - Obstetriz/Obstetra
            - Partero/a no calilficado/a

In [15]:
#Asignamos los tipos de asistentes en la provincia 1 Pichincha filtrada
asisP1 = prov1[prov1['asis_por']=='M�dico/a']
asisP2 = prov1[prov1['asis_por']=='Obstetriz/Obstetra']
asisP3 =prov1[prov1['asis_por']=='Partero/a no calilficado/a']

In [16]:
#Creamos un arreglo de la provincia que vamos a graficar
provincias = ['Pichincha']

#A continuación creamos un arreglo con los asistentes de parto que vamos a representar
asistentes = ['Medico/a','Obstetriz/Obstetra','Partero/a']

#Asiganamos los valores contados de cada provincia en sus respectivos asistentes contados
data = {'provincias':provincias,
         'Medico/a':asisP1.count(),
         'Obstetriz/Obstetra':asisP2.count(),
         'Partero/a':asisP3.count()
    
}

In [17]:
#librerias para la grafica
from bokeh.transform import dodge       #Se ua para obtener un solo valor
from bokeh.core.properties import value #Tomar un valor espececifico de una variable

#Colocamos todo el data, en un ColumnDataSource llamada fuente 
fuente = ColumnDataSource(data)

#Establecemos de los valores a graficar y caracteriscas de la grafica
#rango de x = provincia, y= hasta 350 
grafica_barra = figure(x_range=provincias, y_range=(0, 350), plot_height=350, title="Asistentes en Partos de Pichincha")

#Tipo de glifo .vbar, establecemos sus caracteristicas de la barra
#Posicionamos las barras para que no se monten
#En el dodge asignamos = un nombre de campo para configurar DataSpec, el desplazamiento fijo de la columna, rango calcular las coordenadas
grafica_barra.vbar(x = dodge('provincias', -0.25, range=grafica_barra.x_range), top='Medico/a', width=0.2, source=fuente,
       color="#FC3E3E", legend = value("Médico/a"))

grafica_barra.vbar(x = dodge('provincias',  0.0,  range=grafica_barra.x_range), top='Obstetriz/Obstetra', width=0.2, source=fuente,
       color="#02BEBB", legend = value("Obstetriz/Obstetra"))

grafica_barra.vbar(x = dodge('provincias',  0.25, range=grafica_barra.x_range), top='Partero/a', width=0.2, source=fuente,
       color="#289C2F", legend = value("Partero/a no calilficado/a"))


grafica_barra.x_range.range_padding = 0       #Ancho de la barra
grafica_barra.xgrid.grid_line_color = None    #Quitamos linea vertical
grafica_barra.legend.location = "top_left"    #Posicion para indicador de datos
grafica_barra.legend.orientation = "vertical"   #Colocamos orientacion horizontal
show(grafica_barra)                               #presentamos

#### La mayoría de partos en esta provincia de Pichincha del sector Sierra son atendidos por un Médico con 249, Obstetriz con 35 y Pateros con 1

## Filtración de datos -  Provincia del Guayas

A continuación se realiza una filtración de datos de la Provincia del Guayas ya que es una provincia con mayor población y contiene muchos datos variados en el csv

Hacemos una filtración para tener la variable datos3 que comprende solo Pronvicia del "Guayas", cantón "Guayaquil", y que sean de área "Urbana"

In [18]:
#Filtra 3 veces
datos1 = datos[datos['prov_insc'] == 'Guayas']
datos2 = datos1[datos1['cant_insc'] == 'Guayaquil']
datos3 = datos2[datos2['area_nac'] == 'Urbana']
datos3
#Presentamos los datos 3

Unnamed: 0            tip_insc  ofi_insc prov_insc  cant_insc  \
23            24  Inscripci�n Normal         1    Guayas  Guayaquil   
24            25  Inscripci�n Normal         1    Guayas  Guayaquil   
25            26  Inscripci�n Normal         1    Guayas  Guayaquil   
30            31  Inscripci�n Normal         1    Guayas  Guayaquil   
32            33  Inscripci�n Normal         1    Guayas  Guayaquil   
36            37  Inscripci�n Normal         1    Guayas  Guayaquil   
41            42  Inscripci�n Normal         1    Guayas  Guayaquil   
42            43  Inscripci�n Normal         1    Guayas  Guayaquil   
49            50  Inscripci�n Normal         1    Guayas  Guayaquil   
50            51  Inscripci�n Normal         1    Guayas  Guayaquil   
57            58  Inscripci�n Tard�a         1    Guayas  Guayaquil   
68            69  Inscripci�n Normal         1    Guayas  Guayaquil   
70            71  Inscripci�n Normal         1    Guayas  Guayaquil   
72            73  Inscripci�n Normal         1    Guayas  Guayaquil   
74            75  Inscripci�n Normal         1    Guayas  Guayaquil   
78            79  Inscripci�n Normal         1    Guayas  Guayaquil   
79            80  Inscripci�n Normal         1    Guayas  Guayaquil   
83            84  Inscripci�n Normal         1    Guayas  Guayaquil   
85            86  Inscripci�n Normal         1    Guayas  Guayaquil   
88            89  Inscripci�n Normal         1    Guayas  Guayaquil   
91            92  Inscripci�n Normal         1    Guayas  Guayaquil   
93            94  Inscripci�n Normal         1    Guayas  Guayaquil   
94            95  Inscripci�n Normal         1    Guayas  Guayaquil   
100          101  Inscripci�n Normal         1    Guayas  Guayaquil   
103          104  Inscripci�n Normal         1    Guayas  Guayaquil   
105          106  Inscripci�n Normal         1    Guayas  Guayaquil   
106          107  Inscripci�n Normal         1    Guayas  Guayaquil   
107          108  Inscripci�n Normal         1    Guayas  Guayaquil   
111          112  Inscripci�n Normal         1    Guayas  Guayaquil   
113          114  Inscripci�n Normal         1    Guayas  Guayaquil   
...          ...                 ...       ...       ...        ...   
2323        2324  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2324        2325  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2325        2326  Inscripci�n Normal         1    Guayas  Guayaquil   
2330        2331  Inscripci�n Normal         1    Guayas  Guayaquil   
2332        2333  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2334        2335  Inscripci�n Normal         1    Guayas  Guayaquil   
2337        2338  Inscripci�n Normal         1    Guayas  Guayaquil   
2341        2342  Inscripci�n Normal         1    Guayas  Guayaquil   
2347        2348  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2350        2351  Inscripci�n Normal         1    Guayas  Guayaquil   
2360        2361  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2361        2362  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2363        2364  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2364        2365  Inscripci�n Normal         1    Guayas  Guayaquil   
2365        2366  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2366        2367  Inscripci�n Normal         1    Guayas  Guayaquil   
2378        2379  Inscripci�n Normal         1    Guayas  Guayaquil   
2381        2382  Inscripci�n Normal         1    Guayas  Guayaquil   
2390        2391  Inscripci�n Normal         1    Guayas  Guayaquil   
2436        2437  Inscripci�n Normal         1    Guayas  Guayaquil   
2437        2438  Inscripci�n Normal         1    Guayas  Guayaquil   
2438        2439  Inscripci�n Normal         1    Guayas  Guayaquil   
2439        2440  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2440        2441  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2441        2442  Inscripci�n 

### Se presenta datos por estado Civil de las madres en la provincia del Guayas del area urbana

In [19]:
datos3.set_index('est_civil')#se establece lugares donde ocurrió en Nacimiento

Unnamed: 0            tip_insc  ofi_insc prov_insc  cant_insc  \
est_civil                                                                  
Unida              24  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             25  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             26  Inscripci�n Normal         1    Guayas  Guayaquil   
Soltera            31  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              33  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             37  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              42  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              43  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             50  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             51  Inscripci�n Normal         1    Guayas  Guayaquil   
Soltera            58  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Soltera            69  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              71  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              73  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             75  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              79  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             80  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              84  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              86  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             89  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              92  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada             94  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida              95  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida             101  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida             104  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida             106  Inscripci�n Normal         1    Guayas  Guayaquil   
Soltera           107  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada            108  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida             112  Inscripci�n Normal         1    Guayas  Guayaquil   
Soltera           114  Inscripci�n Normal         1    Guayas  Guayaquil   
...               ...                 ...       ...       ...        ...   
Se ignora        2324  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Soltera          2325  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Soltera          2326  Inscripci�n Normal         1    Guayas  Guayaquil   
Soltera          2331  Inscripci�n Normal         1    Guayas  Guayaquil   
Se ignora        2333  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Casada           2335  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida            2338  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada           2342  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida            2348  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Casada           2351  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida            2361  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Soltera          2362  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Unida            2364  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Casada           2365  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada           2366  Inscripci�n Tard�a         1    Guayas  Guayaquil   
Soltera          2367  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida            2379  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida            2382  Inscripci�n Normal         1    Guayas  Guayaquil   
Casada           2391  Inscripci�n Normal         1    Guayas  Guayaquil   
Unida            2437  Inscripci�n Normal         1    Guay

Las categorias para estado civil de las madres son: 
            - Soltera
            - Casada
            - Unida
            - Divorciada
            - Se Ingnora
            - Separada

In [37]:
#Libreria para la grafica elegida
#El row permite que todos los objetos tengan el mismo modo de tamaño, que se requiere para que funcionen los diseños complejos.
from bokeh.layouts import row
#Presentamos la cantidad estados civil de las madres de los partos

#Agrupamos en un nuevoDateFrame
nuevo = datos3.groupby('est_civil', as_index = False).count()

source = ColumnDataSource(nuevo)
#Creamos objeto graf_EC tipo figure con metodos para agregar diferentes tipos de glifos a un gráfico.
graf_EC = figure(title="Estado Civil de las Madres", toolbar_location=None,
            y_range=nuevo['est_civil'], x_range=[0,231], tools="")

#Caracteristicas de la lineas
graf_EC.segment(x0=0, y0="est_civil" , x1="tip_insc", y1 ="est_civil", line_width=2, line_color="green",source = source )

#Caracteristicas de los circulos
graf_EC.circle(x="tip_insc", y="est_civil", size=15, fill_color="red", line_color="green", line_width=3, source=source )

graf_EC.add_tools(HoverTool(tooltips=[("Estados:","@tip_insc")]))

show(row(graf_EC, sizing_mode = "scale_width"))#Presentamos la grafica, utilizamos el scale-width para que ocupe todo el ancho del casillero

ValueError: Unrecognized range input: '[0, 'tip_insc']'

#### Está gráfica refleja los estados civiles de las madres que dan a luz, la mayoría de madres son Solteras o están en Unión libre.

Ahora establecemos como indice de la tabla al atributo "lugar_ocur", el cual es el lugar de nacimiento de los bebés.

In [21]:
datos3.set_index('lugar_ocur')#se establece lugares donde ocurrió en Nacimiento

Unnamed: 0            tip_insc  \
lugar_ocur                                                                    
Establecimiento de la Junta de Beneficencia          24  Inscripci�n Normal   
Hospital, Cl�nica o Consultorio Privado              25  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          26  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          31  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          33  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          37  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          42  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          43  Inscripci�n Normal   
Hospital, Cl�nica o Consultorio Privado              50  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          51  Inscripci�n Normal   
Establecimiento del Ministerio de Salud              58  Inscripci�n Tard�a   
Establecimiento de la Junta de Beneficencia          69  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          71  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          73  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          75  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          79  Inscripci�n Normal   
Establecimiento del Ministerio de Salud              80  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          84  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          86  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          89  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          92  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          94  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia          95  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia         101  Inscripci�n Normal   
Hospital, Cl�nica o Consultorio Privado             104  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia         106  Inscripci�n Normal   
Establecimiento del Ministerio de Salud             107  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia         108  Inscripci�n Normal   
Hospital, Cl�nica o Consultorio Privado             112  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia         114  Inscripci�n Normal   
...                                                 ...                 ...   
Hospital, Cl�nica o Consultorio Privado            2324  Inscripci�n Tard�a   
Establecimiento del Ministerio de Salud            2325  Inscripci�n Tard�a   
Establecimiento del Ministerio de Salud            2326  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia        2331  Inscripci�n Normal   
Establecimiento del Ministerio de Salud            2333  Inscripci�n Tard�a   
Establecimiento de la Junta de Beneficencia        2335  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia        2338  Inscripci�n Normal   
Establecimiento de la Junta de Beneficencia        2342  Inscripci�n Normal   
Establecimiento del Ministerio de Salud            2348  Inscripci�n Tard�a   
Establecimiento de la Junta de Beneficencia        2351  Inscripci�n Normal   
Hospital, Cl�nica o Consultorio Privado            2361  Inscripci�n Tard�a   
Establecimiento de la Junta de Beneficencia        2362  Inscripci�n Tard�a   
Hospital, Cl�nica o Consultorio Privado            2364  Inscripci�n Tard�a   
Establecimiento del Ministerio de Salud            2365  Inscripci�n Normal   
Establecimiento del IESS                           2366  Inscripci�n Tard�a   
Establecimiento del Ministerio de Salud            2367  Inscripci�n Normal   
Establecimiento del Ministerio de Salud            2379  Inscripci�n Normal   
Hospital, Cl�nica o Consultorio Privado            2382  Inscripci�n Normal   
Establecimiento 

Las categorías para lugar de Nacimiento son:
            - Establecimiento de la Junta de Beneficencia
            - Hospital, Clínica o Consultorio Privado
            - Establecimiento del Ministerio de Salud
            - Establecimiento del IESS
            - Otro Establecimiento Público
            - Casa

In [22]:
#Creamos una varible juntaB y almacenamos la cantidad de nacidos en Juntas de Beneficencia
juntaB = datos3[datos3['lugar_ocur']=='Establecimiento de la Junta de Beneficencia']
juntaB.count().head(1)#Realizamos conteo y presentamos la cantidad con un head()

Unnamed: 0    239
dtype: int64

In [23]:
#Creamos una varible Hc y almacenamos la cantidad de nacidos en Hospitales o Clinicas
Hc= datos3[datos3['lugar_ocur']=='Hospital, Cl�nica o Consultorio Privado']
Hc.count().head(1)#Realizamos conteo y presentamos la cantidad con un head()

Unnamed: 0    165
dtype: int64

In [24]:
#Creamos una varible Em y almacenamos la cantidad de nacidos en Establecimientos de Ministerios de Salud
Em = datos3[datos3['lugar_ocur']=='Establecimiento del Ministerio de Salud']
Em.count().head(1)#Realizamos conteo y presentamos la cantidad con un head()

Unnamed: 0    195
dtype: int64

In [25]:
#Creamos una varible EIess y almacenamos la cantidad de nacidos en Establecimientos del Iess
EIess = datos3[datos3['lugar_ocur']=='Establecimiento del IESS']
EIess.count().head(1)#Realizamos conteo y presentamos la cantidad con un head()

Unnamed: 0    24
dtype: int64

In [26]:
#Creamos una varible oPublic y almacenamos la cantidad de nacidos en Establecimientos publicos
oPublic = datos3[datos3['lugar_ocur']=='Otro Establecimiento P�blico']
oPublic.count().head(1)#Realizamos conteo y presentamos la cantidad con un head()

Unnamed: 0    3
dtype: int64

In [27]:
#Creamos una varible casa y almacenamos la cantidad de nacidos en Casas
casa = datos3[datos3['lugar_ocur']=='Casa']
casa.count().head(1)#Realizamos conteo y presentamos la cantidad con un head()

Unnamed: 0    2
dtype: int64

### Gráfica circular de datos por lugar de nacimiento de los bebes

In [28]:
#librerias para la grafica
from math import pi                    #Para calcular el tamaño del circulo
from bokeh.palettes import Viridis     #Colores para las divisones
from bokeh.transform import cumsum     #Devuelve un marco de datos o una serie del mismo tamaño que contiene la suma acumulada

#Almacenamos los datos a represantar en X
x = {
    'Establecimiento de la Junta de Beneficencia': 239,
    'Hospital, Clínica o Consultorio Privado': 165,
    'Establecimiento del Ministerio de Salud': 195,
    'Establecimiento del IESS': 24,
    'Otro Establecimiento Público':3,
    'Casa':2
}


data = pd.Series(x).reset_index(name='value').rename(columns={'index':'cant'})
#Para el tamaño de los ángulos de cada categoria
data['angle'] = data['value']/data['value'].sum() * 2*pi
#Colores del circulo con rango del conjunto X, toma la cantidad de colores que tenga x
data['color'] = Viridis[len(x)]

#Creamos objeto graf_Lugar para graficar -  establecemos características
graf_Lugar = figure(plot_width=700, plot_height=350, title="Lugares de Nacimientos", toolbar_location=None,
           tools="hover", tooltips="@cant: @value", x_range=(-0.5, 1.5))

#Tipo de glifo .wedge, establecemos sus caracteristicas del pastel
graf_Lugar.wedge(x=0, y=1, radius=0.4, start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),
        line_color="black", fill_color='color', legend='cant', source=data)


graf_Lugar.axis.axis_label = None         #Quitamos cambios del eje x
graf_Lugar.axis.visible = False           #Ocultamos el eje X y Y
graf_Lugar.grid.grid_line_color = None    #Ocultamos las lineas horizontales y verticales de la grafica

show(graf_Lugar)                          #Presentamos la grafica

### Filtración de datos, se agrega las madres de Estado civil: Solteras

Se almacena en la variable datos4 la informacion filtrada de datos3, agregandole que solo obtenga alas madres "SOLTERAS"

In [29]:
#Filtracion por estado civil = Soltera
datos4 = datos3[datos3['est_civil'] == 'Soltera']
datos4
#Presentamos el nuevo conjunto de datos

Unnamed: 0            tip_insc  ofi_insc prov_insc  cant_insc  \
30            31  Inscripci�n Normal         1    Guayas  Guayaquil   
57            58  Inscripci�n Tard�a         1    Guayas  Guayaquil   
68            69  Inscripci�n Normal         1    Guayas  Guayaquil   
106          107  Inscripci�n Normal         1    Guayas  Guayaquil   
113          114  Inscripci�n Normal         1    Guayas  Guayaquil   
118          119  Inscripci�n Normal         1    Guayas  Guayaquil   
127          128  Inscripci�n Tard�a         1    Guayas  Guayaquil   
130          131  Inscripci�n Normal         1    Guayas  Guayaquil   
140          141  Inscripci�n Normal         1    Guayas  Guayaquil   
141          142  Inscripci�n Normal         1    Guayas  Guayaquil   
146          147  Inscripci�n Normal         1    Guayas  Guayaquil   
166          167  Inscripci�n Normal         1    Guayas  Guayaquil   
175          176  Inscripci�n Normal         1    Guayas  Guayaquil   
179          180  Inscripci�n Normal         1    Guayas  Guayaquil   
191          192  Inscripci�n Normal         1    Guayas  Guayaquil   
212          213  Inscripci�n Normal         1    Guayas  Guayaquil   
219          220  Inscripci�n Normal         1    Guayas  Guayaquil   
249          250  Inscripci�n Normal         1    Guayas  Guayaquil   
262          263  Inscripci�n Normal         1    Guayas  Guayaquil   
301          302  Inscripci�n Normal         1    Guayas  Guayaquil   
304          305  Inscripci�n Normal         1    Guayas  Guayaquil   
310          311  Inscripci�n Normal         1    Guayas  Guayaquil   
362          363  Inscripci�n Normal         1    Guayas  Guayaquil   
363          364  Inscripci�n Normal         1    Guayas  Guayaquil   
370          371  Inscripci�n Normal         1    Guayas  Guayaquil   
390          391  Inscripci�n Normal         1    Guayas  Guayaquil   
392          393  Inscripci�n Normal         1    Guayas  Guayaquil   
404          405  Inscripci�n Normal         1    Guayas  Guayaquil   
436          437  Inscripci�n Normal         1    Guayas  Guayaquil   
443          444  Inscripci�n Normal         1    Guayas  Guayaquil   
...          ...                 ...       ...       ...        ...   
2173        2174  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2174        2175  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2175        2176  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2178        2179  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2183        2184  Inscripci�n Normal         1    Guayas  Guayaquil   
2184        2185  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2192        2193  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2211        2212  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2213        2214  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2214        2215  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2226        2227  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2228        2229  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2230        2231  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2253        2254  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2254        2255  Inscripci�n Normal         1    Guayas  Guayaquil   
2258        2259  Inscripci�n Normal         1    Guayas  Guayaquil   
2259        2260  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2267        2268  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2285        2286  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2286        2287  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2287        2288  Inscripci�n Normal         1    Guayas  Guayaquil   
2303        2304  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2313        2314  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2324        2325  Inscripci�n Tard�a         1    Guayas  Guayaquil   
2325        2326  Inscripci�n 

### Grafica de etnias 

Con los datos filtrados, graficamos la cantidad de etnias en el país en el año 2014, de las cuales tenemos las siguientes categorías:
            - Mestiza
            - Blanca
            - Afroecuatoriana/Afrodescendiente
            - Mulata
            - Negra
            - Montubia
            - Otra
            - Ignorado

In [30]:
#Presentamos la cantidad de Datos por etnias del Guayas
etnias = list((datos4["etnia"]))#Creamos una lista con que contiene todos las etnias
uniEt = list(set(etnias))#Asigamos la lista a otra variable uniEt

#Contamos con la funcion .count() numero de nacimientos por cada categoria de etnia y se asigna a cada variable una posición
et1 = etnias.count(uniEt[0])
et2 = etnias.count(uniEt[1])
et3 = etnias.count(uniEt[2])
et4 = etnias.count(uniEt[3])
et5 = etnias.count(uniEt[4])
et6 = etnias.count(uniEt[5])
et7 = etnias.count(uniEt[6])
et8 = etnias.count(uniEt[7])

#Agregamos a una lista_et todas las etnias contabilizadas
lista_et=[et1,et2,et3,et4,et5,et6,et7,et8]

#Creamos objeto graf_etnias tipo Figure y establecemos el rango de Y que tomara las etnias de forma individual
graf_etnias = figure(y_range=uniEt, x_range=[0,20])

#Para graficar utilizamos el glifo .circle y le añadimos caracteristicas 
#X = lista_Et (cantidades contadas) , Y = uniEt (categorias de etnias)
graf_etnias.circle(lista_et, uniEt, size=10, fill_color="#C1E340", line_color="brown", line_width=3)

show(graf_etnias)#Presentamos la grafica

In [31]:
#Cantidades representadas por etnias
et1,et2,et3,et4,et5,et6,et7,et8

(177, 1, 1, 1, 2, 3, 5, 6)

#### Se puede apreciar que la etnia en la que más hay nacimientos es la etnia mestiza y negra ya que estas etnias son las más pobladas ecuador

Ahora para visualizar de mejor manera las categorías del atributo Tipos de Inscripción, establecemos como indice de la tabla al atributo "tip_insc" y visualizamos sus categorias para filtrarlas

In [32]:
#Establecemos como indice tipos de inscripcion
datos4.set_index('tip_insc')

Unnamed: 0  ofi_insc prov_insc  cant_insc   parr_insc  \
tip_insc                                                                    
Inscripci�n Normal          31         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Tard�a          58         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal          69         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         107         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         114         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         119         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Tard�a         128         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         131         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         141         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         142         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         147         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         167         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         176         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         180         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         192         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         213         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         220         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         250         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         263         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         302         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         305         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         311         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         363         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         364         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         371         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         391         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         393         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         405         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         437         1    Guayas  Guayaquil      Tarqui   
Inscripci�n Normal         444         1    Guayas  Guayaquil      Tarqui   
...                        ...       ...       ...        ...         ...   
Inscripci�n Tard�a        2174         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2175         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2176         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2179         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Normal        2184         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2185         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2193         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2212         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2214         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2215         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2227         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2229         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2231         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2254         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Normal        2255         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Normal        2259         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2260         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2268         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard�a        2286         1    Guayas  Guayaquil  Rocafuerte   
Inscripci�n Tard

Se obtuvo 196 datos de los cuales las categorias que se identifica son:
            - Inscripción Normal
            - Inscripción Tardía

In [33]:
#Creamos una varible i_Normal y filtramos por tipo de Inscricpción Normal
i_Normal = datos4[datos4['tip_insc']=='Inscripci�n Normal']
i_Normal.count().head(2)#Realizamos conteo de esa inscripción y presentamos la cantidad con un head()

Unnamed: 0    149
tip_insc      149
dtype: int64

In [34]:
#Creamos una varible i_Tardia y filtramos por tipo de Inscricpción Tardia
i_Tardia = datos4[datos4['tip_insc']=='Inscripci�n Tard�a']
i_Tardia.count().head(2)#Realizamos conteo de esa inscripción y presentamos la cantidad con un head()

Unnamed: 0    47
tip_insc      47
dtype: int64

- Se obtuvo un total de 149 inscripciones Normales y 47 inscripciones Tardías por lo que resultan 196 datos, lo cual esta correcto ya que cuando colocamos como indice a ese atributo obtuvimos 196 datos

### Gráfica por tipo de Inscripciones de los nacimientos de la Provincia del Guayas, sector urbano y de madres Solteras

In [35]:
#Libreria para la gráfica
from bokeh.models import FactorRange #Un rango de valores para una dimensión categórica.

#Categorias a graficar
inscripciones = [
    ("Inscripción Normal", ""),
    ("Inscripción Tardía", "") 

]

#Creamos el objeto graf_Inscrip tipo Figure para asignarle caracteristicas de la grafica
#El Factorange ayuda a obtener el rango de la variable inscripciones
graf_Inscrip = figure(x_range = FactorRange(*inscripciones), plot_height=250, toolbar_location=None, tools="")

#Asignamos los valores obtenidos a cada respectiva Inscripción
ValInscripciones = [149, 47]

#Como es una gráfica de barras utilizamos el glifo .vbar y establecemos las corrdenadas x y el alto de las barras en top
graf_Inscrip.vbar(x = inscripciones, color=["#0FFF83","#F61F46"], top = ValInscripciones, width=0.9, alpha=0.5)

#Para la grafica de la linea utilizamos el .line, le asignamos coordenadas de X y Y y más caracteristicas 
#Colocamos encima de que barras va a ir la linea
graf_Inscrip.line(x=["Inscripción Normal", "Inscripción Tardía"], y=[149,47], color="red", line_width = 3.5)


graf_Inscrip.y_range.start = 0                 #Establecemos el comiezo de los intervalos para el eje y 
graf_Inscrip.x_range.range_padding = 0.2       #Establecemos el tamaño del relleno de la barra
graf_Inscrip.xgrid.grid_line_color = None      #Quitamos la linea vertical 

show(graf_Inscrip)                             #Presentamos la grafica